Library/Homebrew/test/compiler_failure_spec.rb

```diff
@@ -0,0 +1,41 @@
+require "compilers"
+
+describe CompilerFailure do
+  matcher :fail_with do |expected|
+    match do |actual|
+      actual.fails_with?(expected)
+    end
+  end
```

Comment:
Didn't think about it. Seems like `alias_matcher` can only handle predicates without arguments.